File: src/main/java/j/chain_of_resposibility/like/functional/OddSupport.java
package j.chain_of_resposibility.like.functional;

public final class OddSupport {

  public static Support create(String name) {
    return trouble -> {
      if (trouble.getNumber() % 2 == 1) {
        MessagePrinter.done(name, trouble);
        return ControlBreak.of(trouble).withControlBreak(true);
      } else {
        return ControlBreak.of(trouble);
      }
    };
  }
}

File: src/main/scala/s/chain_of_resposibility/like/functional/OddSupport.scala
package s.chain_of_resposibility.like.functional

import s.chain_of_resposibility.Trouble

object OddSupport {
  def create(name: String): PartialFunction[Trouble, Unit] = {
    case trouble if trouble.number % 2 == 1 =>
      MessagePrinter.done(name, trouble)
  }
}
